app/models/User.coco.coffee

```diff
@@ -323,6 +323,21 @@ module.exports = class User extends CocoModel
       return true if new Date() < new Date(maxFree)
     false
 
+  premiumEndDate: ->
+    return null unless @isPremium()
+    if stripe = @get('stripe')
+      return $.t('subscribe.forever') if stripe.free is true
+      return $.t('subscribe.forever') if stripe.sponsorID
+      return $.t('subscribe.forever') if stripe.subscriptionID
+      stripeEnd =  moment(stripe.free) if _.isString(stripe.free)
+      # return moment(stripe.free).utc().format('ll') if _.isString(stripe.free)
+    if products = @get('products')
+      homeProducts = @activeProducts('basic_subscription')
+      endDate = _.max(homeProducts, (p) => new Date(p.endDate)).endDate
+      productsEnd = moment(endDate)
+      return stripeEnd.utc().format('ll') if stripeEnd and stripeEnd.isAfter(productsEnd)
```

Comment:
stripeEnd is actually under if condition above and still has access here, coffee and old js feels 😢

Reply:
oh yeah. let me try fix it